test/qdmi/test_driver.cpp

```diff
@@ -0,0 +1,548 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "qdmi/Driver.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+#include <qdmi/client.h>
+#include <random>
+#include <sstream>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace testing {
+namespace {
+auto stringConcat5(const std::string& a, const std::string& b,
+                   const std::string& c, const std::string& d,
+                   const std::string& e) -> std::string {
+  std::stringstream ss;
+  ss << a << b << c << d << e;
+  return ss.str();
+}
+// NOLINTBEGIN(readability-identifier-naming,cppcoreguidelines-avoid-const-or-ref-data-members)
+MATCHER_P2(IsBetween, a, b,
+           stringConcat5(negation ? "isn't" : "is", " between ",
+                         PrintToString(a), " and ", PrintToString(b))) {
+  return a <= arg && arg <= b;
+}
+// NOLINTEND(readability-identifier-naming,cppcoreguidelines-avoid-const-or-ref-data-members)
+} // namespace
+} // namespace testing
+namespace qc {
+class DriverTest : public testing::TestWithParam<std::string> {
+protected:
+  QDMI_Session session = nullptr;
+  QDMI_Device device = nullptr;
+#ifndef _WIN32
+  static void SetUpTestSuite() {
+    // avoid loading the same dynamic library twice as it would cause an
+    // exception; this only happens, because the `DriveJobTest` inherits from
+    // `DriverTest` and both call `SetUpTestSuite()` when instantiated
+    static bool added = false;
+    if (!added) {
+      qdmi::Driver::get().addDynamicDeviceLibrary(DYN_DEV_LIB, "MQT_NA_DYN");
+      added = true;
+    }
```

Comment:
Should loading the same library twice really cause an exception? Shouldn't this "just work"?